Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -338,10 +347,57 @@ namespace PhysX
     }
 
     // DisplayCallback
-    void EditorHeightfieldColliderComponent::Display(AzFramework::DebugDisplayRequests& debugDisplay) const
+    void EditorHeightfieldColliderComponent::Display(const AzFramework::ViewportInfo& viewportInfo,
+        AzFramework::DebugDisplayRequests& debugDisplay) const
     {
-        const auto& heightfieldConfig = static_cast<const Physics::HeightfieldShapeConfiguration&>(*m_shapeConfig);
-        m_colliderDebugDraw.DrawHeightfield(debugDisplay, m_colliderConfig, heightfieldConfig);
+        AzPhysics::SimulatedBody* simulatedBody = GetSimulatedBody();
+        if (!simulatedBody)
+        {
+            return;
+        }
+
+        AzPhysics::StaticRigidBody* staticRigidBody = azrtti_cast<AzPhysics::StaticRigidBody*>(simulatedBody);
+        if (!staticRigidBody)
+        {
+            return;
+        }
+
+        // Calculate the center of a box in front of the camera - this will be the area to draw
+        const AzFramework::CameraState cameraState = AzToolsFramework::GetCameraState(viewportInfo.m_viewportId);
+        const AZ::Vector3 boundsAabbCenter = cameraState.m_position + cameraState.m_forward * physx_heightfieldDebugDrawDistance * 0.5f;
+
+        AZ::u32 shapeCount = staticRigidBody->GetShapeCount();
+        for (AZ::u32 shapeIndex = 0; shapeIndex < shapeCount; ++shapeIndex)
+        {
+            // Shape::GetGeometry expects the bounding box in the local space
+            const AZStd::shared_ptr<Physics::Shape> shape = staticRigidBody->GetShape(shapeIndex);
+            const AZ::Vector3 bodyPosition = staticRigidBody->GetPosition();
+            const AZ::Vector3 shapeOffset = shape->GetLocalPose().first;
+            const AZ::Vector3 aabbCenterLocal = boundsAabbCenter - bodyPosition - shapeOffset;
+
+            // Create the bounds box of the required size
+            const AZ::Aabb boundsAabb = AZ::Aabb::CreateCenterRadius(aabbCenterLocal, physx_heightfieldDebugDrawDistance);
+            if (!boundsAabb.IsValid())
+            {
+                return;
+            }
+
+            // Extract the heightfield geometry within the bounds
+            AZStd::vector<AZ::Vector3> vertices;
+            AZStd::vector<AZ::u32> indices;
```

Comment:
question: Is it possible to reserve the correct size for vertices/indices here?

Reply:
These are effectively output parameters for `Shape::GetGeometry`, so it's responsibility of the downstream function to return the data in the best way.
In our case `Utils::Geometry::GetHeightFieldGeometry` does reserve the correct size